tests/framework/utils.py

```diff
@@ -601,3 +602,28 @@ def writeln(self, data: str):
         self._file.write(data)
         self._file.write("\n")
         self._file.flush()
+
+
+def run_seccompiler(bpf_path, json_path=None):
```

Comment:
This function seems kind of misplaced to be located in the utils file.

Given that it's a requirement for tests, it could be a pytest fixture.

Reply:
I guess we could move it to host_tools/cargo_build.py, I would avoid as much as possible adding fixtures.

Reply:
Since it's a dependency for this test only, I think it makes sense to use it as a fixture. We don't have to call it explicitly or even import it.

Reply:
> Since it's a dependency for this test only, I think it makes sense to use it as a fixture. We don't have to call it explicitly or even import it.

I see your point @gc-plp. However, it might be personal bias, but I really don't like fixtures 😄 . They make for hard to understand, magical code.
This fixture would also need to be parametrised with the bpf and json file paths, and it would also return `void` (aka nothing). I find it confusing to have a fixture that does not return anything.

I think a plain helper function would work just fine here. I like @dianpopa's suggestion to move the function in cargo_build.py. Is this good enough?

Reply:
moved it to `cargo_build.py`. WDYT?

Reply:
>I see your point @gc-plp. However, it might be personal bias, but I really don't like fixtures smile . They make for hard to understand, magical code.

Fixtures aren't great but they avoid the import mechanism built into python, thus treating tests as black boxes.

> This fixture would also need to be parametrised with the bpf and json file paths, and it would also return void (aka nothing). I find it confusing to have a fixture that does not return anything.

The fixture can return a function.

> I think a plain helper function would work just fine here. I like @dianpopa's suggestion to move the function in cargo_build.py. Is this good enough?

Sure, that works too.